File: lib/features/ct/domain/entities/phieu_chi.dart
// ============================================================================
// Domain Layer - Entities
// Based on UC_HKD_TT88_2021 - CT-02: Lập phiếu chi
// ============================================================================

import 'package:equatable/equatable.dart';

class PhieuChi extends Equatable {
  final String id;
  final String soPhieu;
  final DateTime ngayLap;
  final String nguoiNop;
  final String diaChiNguoiNop;
  final String lyDoNop;
  final int soTien;
  final String soTienBangChu;
  final String chungTuGocKemTheo;
  final String hkdInfoId;
  final String? nhaCungCapId;
  final String kyKeToanId;
  final String trangThai;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const PhieuChi({
    required this.id,
    required this.soPhieu,
    required this.ngayLap,
    required this.nguoiNop,
    required this.diaChiNguoiNop,
    required this.lyDoNop,
    required this.soTien,
    required this.soTienBangChu,
    required this.chungTuGocKemTheo,
    required this.hkdInfoId,
    this.nhaCungCapId,
    required this.kyKeToanId,
    required this.trangThai,
    this.createdAt,
    this.updatedAt,
  });

  PhieuChi copyWith({
    String? id,
    String? soPhieu,
    DateTime? ngayLap,
    String? nguoiNop,
    String? diaChiNguoiNop,
    String? lyDoNop,
    int? soTien,
    String? soTienBangChu,
    String? chungTuGocKemTheo,
    String? hkdInfoId,
    String? nhaCungCapId,
    String? kyKeToanId,
    String? trangThai,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return PhieuChi(
      id: id ?? this.id,
      soPhieu: soPhieu ?? this.soPhieu,
      ngayLap: ngayLap ?? this.ngayLap,
      nguoiNop: nguoiNop ?? this.nguoiNop,
      diaChiNguoiNop: diaChiNguoiNop ?? this.diaChiNguoiNop,
      lyDoNop: lyDoNop ?? this.lyDoNop,
      soTien: soTien ?? this.soTien,
      soTienBangChu: soTienBangChu ?? this.soTienBangChu,
      chungTuGocKemTheo: chungTuGocKemTheo ?? this.chungTuGocKemTheo,
      hkdInfoId: hkdInfoId ?? this.hkdInfoId,
      nhaCungCapId: nhaCungCapId ?? this.nhaCungCapId,
      kyKeToanId: kyKeToanId ?? this.kyKeToanId,
      trangThai: trangThai ?? this.trangThai,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }

  @override
  List<Object?> get props => [
    id,
    soPhieu,
    ngayLap,
    nguoiNop,
    diaChiNguoiNop,
    lyDoNop,
    soTien,
    soTienBangChu,
    chungTuGocKemTheo,
    hkdInfoId,
    nhaCungCapId,
    kyKeToanId,
    trangThai,
    createdAt,
    updatedAt,
  ];
}

File: HkdAccounting.Domain/Entities/PhieuChi.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HkdAccounting.Domain.Entities
{
    /// <summary>
    /// Phiếu chi (Payment Voucher) - CT-02
    /// </summary>
    public class PhieuChi
    {
        public long Id { get; set; }

        [Required, MaxLength(20)]
        public string SoPhieu { get; set; }

        [Required]
        public DateTime NgayChi { get; set; }

        [Required, MaxLength(255)]
        public string NguoiNhan { get; set; }

        [MaxLength(500)]
        public string DiaChi { get; set; }

        [MaxLength(500)]
        public string LyDo { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal SoTien { get; set; }

        [MaxLength(255)]
        public string BangChu { get; set; }

        [Required, MaxLength(20)]
        public string PhuongThucThanhToan { get; set; } = "TIEN_MAT";

        public long? DoiTacId { get; set; }
        public long HkdInfoId { get; set; }

        [MaxLength(20)]
        public string TrangThai { get; set; } = "CHO_DUYET";

        [MaxLength(100)]
        public string NguoiLap { get; set; }

        [MaxLength(100)]
        public string NguoiDuyet { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public virtual DoiTac DoiTac { get; set; }
        public virtual HkdInfo HkdInfo { get; set; }
    }
}
